python/pyarrow/table.pxi

```diff
@@ -1838,6 +1953,36 @@ cdef class Table(_PandasConvertible):
 
         return ordered_dict(entries)
 
+    def to_pylist(self):
+        """
+        Convert the Table to a list of dictionaries.
+
+        Returns
+        -------
+        list
+
+        Examples
+        --------
+        >>> import pyarrow as pa
+        >>> table = pa.table([
+        ...     pa.array([1, 2]),
+        ...     pa.array(["a", "b"])
+        ... ], names=["int", "str"])
+        >>> table.to_pylist()
+        [{'int': [1, 2]}, {'str': ['a', 'b']}]
+        """
+        cdef:
+            size_t i
+            size_t num_columns = self.table.num_columns()
+            list entries = []
+            ChunkedArray column
+
+        for i in range(num_columns):
+            column = self.column(i)
+            entries.append({self.field(i).name: column.to_pylist()})
```

Comment:
`self.itercolumns` should do the trickk for you, also the returned columns has `_name` attribute, so you don't need to look into the field.

Last I suggest we use a list comprehension to speed up the building